immut/priority_queue/priority_queue.mbt

```diff
@@ -46,6 +46,50 @@ test "from_array" {
   inspect(queue.peek(), content="Some(5)")?
 }
 
+pub fn to_array[T : Compare](self : ImmutablePriorityQueue[T]) -> Array[T] {
+  let arr = Array::new()
+  loop self, self.peek() {
+    queue, Some(p) => {
+      arr.push(p)
+      continue queue.pop_exn(), queue.pop_exn().peek()
+    }
+    _, None => ()
```

Comment:
this could be made more efficient by taking advantage of the internals

Reply:
> this could be made more efficient by taking advantage of the internals

what "taking advantage of the internals" mean ?

Reply:
what about:

```moon
pub fn to_array[T : Compare](self : ImmutablePriorityQueue[T]) -> Array[T] {
  let arr : Array[T] = []
  fn go(x : Cont[T]) {
    match x {
      Node(v, child) => {
        arr.push(v)
        child.iter(fn(v) { go(v) }) |> ignore
      }
      Empty => ()
    }
  }

  go(self.body)
  arr.sort_by(fn(x, y) { if x < y { 1 } else { -1 } })
  arr
}
```